Src/AutoFixture/DomainName.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents a domain name.
+    /// </summary>
+    public class DomainName
+    {
+        private string domainName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DomainName"/> class. Throws ArgumentNullException
+        /// if domainName is null. Throws ArgumentException if domainName is empty.
+        /// </summary>
+        public DomainName(string domainName)
+        {
+            if (domainName == null)
+            {
+                throw new ArgumentNullException("domainName");
+            }
+
+            if (domainName.Length == 0)
```

Comment:
Mostly, because the `EmailAddressLocalPart` also didn't allow that. I saw your discussion in the other PR about the RFC implementation, and agree that you shouldn't reimplement the validation logic of the MailAddress class. With that in mind, it's perfectly valid for the `domainName` to be emtpy, and let the `MailAddressGenerator` handle invalid domain names.

I've removed the constraint. Should it also be removed from `EmailAddressLocalPart`?